test/utils/combineReducers.spec.js

```diff
@@ -112,6 +112,17 @@ describe('Utils', () => {
       expect(reducer({counter: 0}, { type: increment }).counter).toEqual(1);
     });
 
+    it('should maintain referential integrity if the reducers it is combining do', () => {
+      const reducer = combineReducers({
+        child(state = {}) {
+          return state;
+        }
+      });
+
+      const initialState = reducer(undefined, '@@INIT');
+      expect(reducer(initialState, { type: 'FOO' })).toBe(initialState);
+    });
+
```

Comment:
I'd like to see more tests.
We need to check that:

* With several reducers, having just one return something different is enough to cause inequality
* With several reducers, having all return the same is enough to maintain equality